raiden/blockchain/filters.py

```diff
@@ -109,59 +110,55 @@ class StatelessFilter(LogFilter):
     Pass latest block_number to get_(new|all)_entries to avoid querying it
     """
 
-    def __init__(self, web3: Web3, filter_params: Dict[str, Any]) -> None:
+    def __init__(
+        self,
+        web3: Web3,
+        from_block: BlockNumber,
+        contract_address: ChecksumAddress,
+        topics: Optional[List[Optional[str]]],
+    ) -> None:
         super().__init__(web3, filter_id=None)
-        self.filter_params: Dict[str, BlockSpecification] = filter_params
-        self._last_block: BlockNumber = BlockNumber(-1)
-        self._lock = Semaphore()
 
-    def _do_get_new_entries(
-        self, from_block: BlockSpecification, to_block: BlockSpecification
-    ) -> List[BlockchainEvent]:
-        filter_params = self.filter_params.copy()
-        filter_params["fromBlock"] = from_block
-        filter_params["toBlock"] = to_block
+        # fix off-by-one
+        last_block = from_block - 1
```

Comment:
I'm not sure about this one, I have to review the user code in the future to make sure this is necessary.